### PR TITLE
Documentation updates following moving build of examples to xcommon-cmake

### PR DESCRIPTION
**Pull request by xross**
- Added doc build to Jenkinsfile
- Quite a few updates to out of date documentation (mostly in examples)

Note, we expect to move the AN* examples (and maintain as AN's proper) and replace with a single, simple example.

### CODE REVIEW

**Review by danielpieczko — APPROVED** [review]
Spotted a couple of small typos, but otherwise looks good.

**Comment by xross** [comment]
@shuchitak can you take a look please? Thanks.

File: examples/AN00132_image_class/doc/rst/AN00132.rst [hunk 1]
```diff
@@ -101,104 +125,116 @@ Looking at this in a more detail you can see the following:
...
+and endpoint setup. It is listed below.
 
 .. literalinclude:: endpoint0.xc
    :start-on: cfgDesc
    :end-on: }
 
-This decriptor is in the format decribed by the USB 2.0 standard. It contains the encoding for the endpoints related to control endpoint 0 and the descriptors that describe the two bulk endpoints for data in and out. It also contains the descriptors for the interrupt endpoint. The bulk endpoints are used for transferring image data and non-image data to adjust device controls. The interrupt endpoint is used to send asynchronous event data such as battery low indication or the removal of the memory card to the host from the device.
+It contains a description of the two bulk endpoints for data in and out. It also contains
+the descriptors for an interrupt endpoint. The bulk endpoints are used for transferring image data
+and non-image data to adjust device controls. The interrupt endpoint is used inform the host of
```
Comment (shuchitak) [inline examples/AN00132_image_class/doc/rst/AN00132.rst:191]:
```suggestion
and non-image data to adjust device controls. The interrupt endpoint is used to inform the host of
```

File: examples/AN00132_image_class/doc/rst/AN00132.rst [hunk 2]
```diff
@@ -101,104 +125,116 @@ Looking at this in a more detail you can see the following:
...
+asynchronous event data such as battery low indication or the removal of the memory card.
 
 USB string descriptors
 ......................
 
-The final descriptor for our still image capture device is the string descriptor which the host machine uses to report to the user when the device is enumerated and when the user queries the device on the host system. This is setup as follows.
+String descriptors provide human readable information for your device and you can configure them
+with your USB product information.. The descriptors are placed in an array as shown in the below code.
```
Comment (shuchitak) [inline examples/AN00132_image_class/doc/rst/AN00132.rst:198]:
```suggestion
with your USB product information. The descriptors are placed in an array as shown in the code below.
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 3]
```diff
@@ -34,87 +39,99 @@ Block diagram
...
+-------------------------------------------------
 
-The application in this note uses the XMOS USB device library and creates a USBTMC device which responds to IVI-VISA compliant host application messages.
-The device implements basic message requests for Bulk-IN and Bulk-OUT endpoints. Interrupt-IN endpoint is optional and is not used in this application. The application integrates an open source SCPI parser library and implements a minimal subset of SCPI commands to communicate with host. More SCPI commands can be easily added to this application by following the steps detailed in the ``Using SCPI library to add more SCPI commands`` appendix section.
+The application in this note uses the XMOS USB device library and creates a USBTMC device which
+responds to IVI-VISA compliant host application messages.
+The device implements basic message requests for Bulk IN and Bulk OUT endpoints.
+Interrupt IN endpoint is optional and is not used in this application.
+The application integrates an open source SCPI parser library and implements a minimal subset of
+SCPI commands to communicate with host. More SCPI commands can be easily added to this application
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:50]:
```suggestion
SCPI commands to communicate with the host. More SCPI commands can be easily added to this application
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 4]
```diff
@@ -34,87 +39,99 @@ Block diagram
...
+by following the steps detailed in the ``Using SCPI library to add more SCPI commands`` appendix
+section.
 
-For the USBTMC device class application example, the system comprises three tasks running on separate logical cores of an xCORE-USB multicore microcontroller. 
+For the USBTMC device class application example, the system comprises three tasks running on
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:54]:
```suggestion
For the USBTMC device class application example, the system comprises of three tasks running on
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 5]
```diff
@@ -123,180 +140,193 @@ The USB ID values used for Vendor ID (VID), Product ID (PID) and device version
...
    :end-on: VENDOR_SPECIFIC_PROTOCOL
 
-These are defined in the USB standard as required in the device description for
-TMC class devices and for configuring them as such with the USB host machine.
-   
 USB Device Descriptor
 .....................
 
-``endpoint0.xc`` is where the standard USB device descriptor is declared for a USBTMC class device. Below is the structure which contains this descriptor. This will be requested by the host when the device is enumerated on the USB bus. This descriptor contains the vendor specific defines described above.
+``endpoint0.xc`` is where the USB device descriptor is declared the device, it is shown below.
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:154]:
```suggestion
``endpoint0.xc`` is where the USB device descriptor is declared by the device, it is shown below.
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 6]
```diff
@@ -123,180 +140,193 @@ The USB ID values used for Vendor ID (VID), Product ID (PID) and device version
...
    :end-on: }
 
-From this descriptor you can see that product, vendor and device firmware revision are all coded into this structure. This will allow the host machine to recognise our USBTMC device when it is connected to the USB bus.
+This descriptor contains the product, vendor and device firmware revision.
 
 USB Configuration Descriptor
 ............................
 
-The USB configuration descriptor is used to configure the device class and the endpoint setup. For the USBTMC device provided in this example the configuration descriptor which is read by the host is as follows.
+The USB configuration descriptor is used to inform the host of the configuration the device class
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:166]:
```suggestion
The USB configuration descriptor is used to inform the host of the configuration, the device class
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 7]
```diff
@@ -123,180 +140,193 @@ The USB ID values used for Vendor ID (VID), Product ID (PID) and device version
...
+Note that `bInterfaceProtocol` value is controlled by the define `USBTMC_SUB_CLASS_SUPPORT` and it
+is set to ``0`` implying that no subclass specification applies and the USBTMC interface is not
+required to have an Interrupt IN endpoint.
 
 USB string descriptors
 ......................
 
-The final descriptor for our USBTMC device is the string descriptor which the host machine uses to report to the user when the device is enumerated and when the user queries the device on the host system. This is setup as follows.
+String descriptors provide human readable information for your device and you can configure them
+with your USB product information.. The descriptors are placed in an array as shown in the below
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:183]:
```suggestion
with your USB product information.. The descriptors are placed in an array as shown in the code below.
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 8]
```diff
@@ -123,180 +140,193 @@ The USB ID values used for Vendor ID (VID), Product ID (PID) and device version
...
+code.
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:184]:
```suggestion
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 9]
```diff
@@ -123,180 +140,193 @@ The USB ID values used for Vendor ID (VID), Product ID (PID) and device version
...
 
 .. literalinclude:: endpoint0.xc
    :start-on: String table
    :end-on: }
 
-The last value corresponds to the iSerialNumber of the device, which should contain a value > 0 to form a valid GUID.
+The `XMOS` USB library will take care of encoding the strings into Unicode and structures the
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:190]:
```suggestion
The `XMOS` USB library will take care of encoding the strings into Unicode and structure the
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 10]
```diff
@@ -123,180 +140,193 @@ The USB ID values used for Vendor ID (VID), Product ID (PID) and device version
...
+
+The function ``Endpoint0()`` contains the code for dealing with device requests made from the host
+to the standard control endpoint 0 which is present in all USB devices.
 
 .. literalinclude:: endpoint0.xc
    :start-on: Endpoint 0 Task
    :end-on: } //if(result == XUD_RES_OKAY)
 
-The control interface class requests are handled using the below function. 
+The control interface class requests are handled using the below function.
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:207]:
```suggestion
The control interface class requests are handled using the function below.
```

File: examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst [hunk 11]
```diff
@@ -123,180 +140,193 @@ The USB ID values used for Vendor ID (VID), Product ID (PID) and device version
...
 
 .. literalinclude:: endpoint0.xc
    :start-on: XUD_Result_t ControlInterfaceClassRequests
    :end-on: return XUD_RES_ERR;
 
-This application implements ``GET_CAPABILITIES`` request and uses it to send our device capabilities to the host.
-
+The application implements ``GET_CAPABILITIES`` request and uses it to send the device capabilities
```
Comment (shuchitak) [inline examples/AN00135_test_and_measurement_class/doc/rst/AN00135.rst:213]:
```suggestion
The application implements the ``GET_CAPABILITIES`` request and uses it to send the device capabilities
```

**Review by shuchitak — APPROVED** [review]
Looks good! Minor spellings etc. corrections suggested.

**Comment by xross** [comment]
> Looks good! Minor spellings etc. corrections suggested.

thanks!

**Pull request merged**